### PR TITLE
Add support for MT 5 game translation (tags00)

### PR DESCRIPTION
**Pull request by Zweihorn**
This Pull Request and comprised commits are aiming to add support for the **MT 5 game translation** by help of the `S("string")` enclosure and a `local S = minetest.get_translator()` stub to each applicable '*.lua' file in the mods repectively.

Please note the about 400 lines of code changed in the Lua code of the existing mods by addition of a `S("string")` enclosure for the translation effort. This should be in compliance to both the MT 5 core `lua_api.txt` and the GNU gettext toolchain procedures hopefully.

The `textdomain` is not set intentionally as this is up to the MT developers, I presume.

>   It is intended to be used in the following way, so that it avoids verbose
>   repetitions of `minetest.translate`:
> 
>       local S = minetest.get_translator(textdomain)
>       S(str, ...)
> 
>   As an extra commodity, if `textdomain` is nil, it is assumed to be "" instead.

The code was briefly tested in a MT 5.0.1 client on macOS. This is my first GitHub branch commit ever and please accept my apologies for any inconvienance in advance.

N.B. My family and me are enjoying the results of your coding efforts at home for a long time now.

I would like to thank all of the MT developers and certainly @kaeza for 'intllib' which gave me the idea and the motivation for this contribution.

[snoopy (Zweihorn)] 2019-05-13

### CODE REVIEW

**Comment by Zweihorn** [comment]
Hm, this is unfortunate but not unexpected. I will look how to improve...

File: README.txt [hunk 1]
```diff
@@ -15,7 +15,7 @@ The Minetest Game github master HEAD is generally compatible with the github
 master HEAD of the Minetest engine.
 
 Additionally, when the Minetest engine is tagged to be a certain version (eg.
-0.4.10), Minetest Game is tagged with the version too.
+5.0.1), Minetest Game is tagged with the version too.
```
Comment (DS-Minetest) [inline README.txt]: Why do you change this?

File: mods/dye/init.lua [hunk 2]
```diff
@@ -1,23 +1,28 @@
+-- dye/init.lua
+
 dye = {}
 
+-- Load support for MT game translation.
+local S = minetest.get_translator()
```
Comment (DS-Minetest) [inline mods/dye/init.lua]:
```suggestion
local S = minetest.get_translator()
```
This is one of the warnings of travis.

File: mods/vessels/init.lua [hunk 3]
```diff
@@ -1,6 +1,12 @@
+-- vessels/init.lua
+
 -- Minetest 0.4 mod: vessels
 -- See README.txt for licensing and other information.
 
+-- Load support for MT game translation.
+local S = minetest.get_translator()
+
```
Comment (DS-Minetest) [inline mods/vessels/init.lua]:
```suggestion

```
This is elsewhere, too. See travis.

File: README.txt (hunk 1 above)
Comment (Zweihorn) [reply README.txt]: Taken from stable-5 branch I presumed this appropriate

File: mods/dye/init.lua (hunk 2 above)
Comment (Zweihorn) [reply mods/dye/init.lua]: THX

File: mods/vessels/init.lua (hunk 3 above)
Comment (Zweihorn) [reply mods/vessels/init.lua]: covered by   c0203a3 + 23c20c3 

THX

**Comment by Zweihorn** [comment]
I am neutral if this is 0.4.10 or 5.0.1 and happy to accept.

All checks have passed. Hip hip hooray...

@Calinou - Thx for the warm welcome.

File: mods/fireflies/init.lua [hunk 4]
```diff
@@ -113,14 +118,14 @@ minetest.register_craft( {
 	recipe = {
 		{"farming:string", "farming:string"},
 		{"farming:string", "farming:string"},
-		{"group:stick", ""}
+		{"default:stick", ""}
```
Comment (SmallJoker) [inline mods/fireflies/init.lua]: Unrelated change, please undo.

File: mods/flowers/init.lua [hunk 5]
```diff
@@ -236,7 +240,8 @@ function flowers.mushroom_spread(pos, node)
 	end
 	local pos2 = positions[math.random(#positions)]
 	pos2.y = pos2.y + 1
-	if minetest.get_node_light(pos2, 0.5) <= 3 then
+	if minetest.get_node_light(pos, 0.5) <= 3 and
+			minetest.get_node_light(pos2, 0.5) <= 3 then
```
Comment (SmallJoker) [inline mods/flowers/init.lua]: Unrelated changes here as well, please undo.

File: mods/fireflies/init.lua (hunk 4 above)
Comment (Zweihorn) [reply mods/fireflies/init.lua]: Obviously, I have some pollution in the fromer original code. Bad boy I am.

THX and will depollute...

File: mods/flowers/init.lua (hunk 5 above)
Comment (Zweihorn) [reply mods/flowers/init.lua]: THX and will depollute...

**Comment by Zweihorn** [comment]
THX to @SmallJoker for pointing out the pollutions.

File: mods/fire/init.lua [hunk 6]
```diff
@@ -337,7 +343,7 @@ if fire_enabled then
 		interval = 5,
 		chance = 18,
 		catch_up = false,
-		action = function(pos)
+		action = function(pos, node, active_object_count, active_object_count_wider)
```
Comment (SmallJoker) [inline mods/fire/init.lua]: Also on L329: Unrelated changes, please undo.

File: mods/flowers/init.lua [hunk 7]
```diff
@@ -46,6 +51,7 @@ local function add_simple_flower(name, desc, box, f_groups)
 		paramtype = "light",
 		walkable = false,
 		buildable_to = true,
+		stack_max = 99,
```
Comment (SmallJoker) [inline mods/flowers/init.lua]: Unrelated changes. Did you copy & paste the files from an old MTG version? Please use rebase if you run into conflicts.

File: mods/flowers/init.lua [hunk 8]
```diff
@@ -221,10 +227,8 @@ minetest.register_node("flowers:mushroom_brown", {
 -- Mushroom spread and death
 
 function flowers.mushroom_spread(pos, node)
-	if minetest.get_node_light(pos, 0.5) > 3 then
-		if minetest.get_node_light(pos, nil) == 15 then
-			minetest.remove_node(pos)
-		end
+	if minetest.get_node_light(pos, nil) == 15 then
+		minetest.remove_node(pos)
```
Comment (SmallJoker) [inline mods/flowers/init.lua]: Unrelated changes, please undo.

File: mods/map/init.lua [hunk 9]
```diff
@@ -65,7 +71,7 @@ minetest.register_craftitem("map:mapping_kit", {
 minetest.register_craft({
 	output = "map:mapping_kit",
 	recipe = {
-		{"default:glass", "default:paper", "group:stick"},
+		{"default:glass", "default:paper", "default:stick"},
```
Comment (SmallJoker) [inline mods/map/init.lua]: Ditto

Comment (Zweihorn) [reply mods/map/init.lua]: Oh this is bad. I was aware of some unrelated changes or deviations earlier and tried to fix this in advance. Obviously, I got it the wrong way and one has to presume some more pollutions.

Please accept my apologies and I presume to wait until you reach 'xpane' apparently.

File: mods/flowers/init.lua (hunk 7 above)
Comment (Zweihorn) [reply mods/flowers/init.lua]: should have been the stable-5 branch but I may be wrong

Comment (SmallJoker) [reply mods/flowers/init.lua]: Yes, the `master` branch would be up-to-date. There were already a few little changes since stable-5.

Comment (Zweihorn) [reply mods/flowers/init.lua]: I made fresh start with a pull from master but copied changed files from my earlier stable-5 work into that. Obviously, my scan and check to repair the deviations was a partial failure.

I will learn from this and seek to improve. Thanks again.

File: mods/map/init.lua (hunk 9 above)
Comment (SmallJoker) [reply mods/map/init.lua]: No problem. I'm glad you put effort into this PR, even though Git is not easy to understand at the beginning.

File: mods/default/trees.lua [hunk 10]
```diff
@@ -560,9 +566,12 @@ function default.sapling_on_place(itemstack, placer, pointed_thing,
 			interval) then
 		minetest.record_protection_violation(pos, player_name)
 		-- Print extra information to explain
+--		minetest.chat_send_player(player_name,
+--			itemstack:get_definition().description .. " will intersect protection " ..
```
Comment (Zweihorn) [inline mods/default/trees.lua]: What ist this pollution?  I am going nuts...

Comment (Zweihorn) [reply mods/default/trees.lua]: Oh calm down old boy. This is just th eold code made comments for documentray purposes intentionally by myself.

**Comment by Zweihorn** [comment]
IMHO this commit may be ready for approval.

The code should be called depolluted. THX to @SmallJoker and @DS-Minetest for the checks and their advice.

**Comment by Zweihorn** [comment]
I will call it a day and a preliminary end to my commits now. I would hope to have covered most if not all user-facing strings.

IMHO this Pull Request may be ready for further discussion or approval.

**Comment by Zweihorn** [comment]
Due to a basic misunderstanding I was not aware of the 'luachecks' avalilability for Mac OS. Only recently I installed the 'luarocks' with MacPorts and thus the 'luarocks install luachecks' for code checks before commits.

Please accept my apologies and I will seek do always perform a local code check before commiting in the furture.

**Comment by Zweihorn** [comment]
**(1) Add function `mod.get_translator` to any mod with serveral lua files.**

Presumably, setting the `textdomain` is a task to the developers as mentioned in the top comment. This task could be helped and more easily accomplished when once required per mod only. Thus only the main lua file loads the `minetest.get_translator()` and any sub-ordinated lua file should reuse the then defined `mod.get_translator` subsequently.

Example:

(a) main lua file:  `mods/beds/init.lua`
```
-- Load support for MT game translation.
local S = minetest.get_translator()
. . . 
beds = { }
beds.get_translator = S
```
(b) sub-ordinated lua file: `mods/beds/beds.lua`
```
-- support for MT game translation.
local S = beds.get_translator
```
Until further notice this is supposed as redundant if there is solely the ini.lua file in a mod respectively.

**(2) luacheck and test with MT client**

- Total: 0 warnings / 0 errors in 59 files
- No apparent issues after loading into a MT 5.0.1 client

Hope this helps.

**Comment by Zweihorn** [comment]
Please note and consider my current position:

Log functions like `minetest.log` are aimed at providing information on technical issues and any respective strings should not be tramslated to avoid disambiguations. A translated log message may lead to confusion between users and developers if copied into problem reports to an international project team like MT devs.

Excerpt from `vessels/init.lua` lines 75-77 as an example:
```
on_metadata_inventory_take = function(pos, listname, index, stack, player)
	minetest.log("action", player:get_player_name() ..
		" takes stuff from vessels shelf at ".. minetest.pos_to_string(pos))
```
Hope this helps and could find a consent. Any advice welcome.

File: mods/flowers/init.lua (hunk 8 above)
Comment (Zweihorn) [reply mods/flowers/init.lua]: should be OK now

**Comment by Zweihorn** [comment]
Resolved and continued by #2368 Pull Request.

**Comment by Linuxdirk** [comment]
> The textdomain is not set intentionally

Textdomain = mod name is suggested. Why not simply do it like this?

**Comment by Wuzzy2** [comment]
Umm, why do we have multiple PRs for translations? o_O I just wrote a huge review of https://github.com/minetest/minetest_game/pull/2466. I am not going to write yet another review …
